app/controllers/case_logs_controller.rb

```diff
@@ -107,9 +107,13 @@ def responses_for_page(page)
         day = params["case_log"]["#{question_key}(3i)"]
         month = params["case_log"]["#{question_key}(2i)"]
         year = params["case_log"]["#{question_key}(1i)"]
-        next unless day.present? && month.present? && year.present?
+        next unless [day, month, year].any?(&:present?)
```

Comment:
Don't you need all of these?

Reply:
Yeah, so that gets checked on the next line. This check right here is to see if any fields at all have been filled in. This indicates that there was an attempt to fill out the date. After that we check if the values were valid and set it to a fake (01, 01, 0) date in order to display the error.

We wouldn't want it to fail if there was no attempt to fill in the fields to begin with, but we'd want it to fail if only 1 or 2 fields are filled in instead of just not saving it silently

Reply:
Oh yeah, of course. Thanks